docs/developer/style_guide.md

```diff
@@ -0,0 +1,579 @@
+compilation for all supported compilers once submitted as a merge request into
+the Bitbucket repository, but these should be performed beforehand to minimise
+the effort in bugfixing.
+
+
+### File handling
+
+One obvious area of machine dependency is that of file handling. For this reason,
+the `io` module should be used to handle the opening of all files. The programmer
+is responsible for reading / writing the data and closing the file. Files are
+not to be kept open for any longer than strictly necessary.
+
+
+### Compiler/preprocessor directives
+
+These should be avoided wherever possible as they hinder portability. They should
+only be used in the `Utility` modules for machine specific code. The one exception
+to this rule, is the use of a single `#ifdef debug` block at the start
+of each routine to check for conditions of usage. For this reason, all module
+filenames should be given the extension `.F90` not `.f90` to enable
+preproccesing on the widest range of compilers available.
+
+
+## Fortran90 Features
+
+
+### Use of dynamic arrays
+
+Each routine is responsible for allocating/using/deallocating its own workspace.
+We recommend that such `allocate`/`deallocate`s only occur once in a routine -
+never inside a loop as this will produce a significant performance penalty. You
+must ensure that you always `deallocate` local work arrays on termination of the
+routine. Obviously, care needs to be taken if there are multiple exit points
+from a routine.
+
+
+### Pointers
+
+Always accept passed arrays in subprograms as arrays and not as pointers. This
+will force automatic dereferencing of the array and enable better optimisations
+by the compiler.
+
+
+### Flow of control
+
+Use of constructs such as `select case` and `do ... end do`, with labelling of
+"do loops", etc. is recommended for clarity. Use of `cycle` and `exit` is OK -
+but only use to move a maximum of one level of nesting where possible. Do NOT
+use `goto`.
+
+
+### Functions
+
+A function subprogram should be "pure", i.e. generate no side-effects through
+modification of variables in argument list or module variables through "use
+association". Avoid recursion.
+
+
+### Use of `use`
+
+Use modules with an `only` clause if only a small number of variables
+required. Put appropriate `use` statements into each subroutine required, NOT
+into the module header, in order to keep track of module variable modification.
+
+
+### Line numbers
+
+Avoid where possible. Line numbers are only to be used for format statements,
+and `end=` or `err=` in file handling. Do not use `continue`.
+
+
+### General features
+
+Avoid multiple statement lines unless trivially simple.
+
+
+## Modules
+
+
+### General
+
+All code is to be contained within modules.
+
+
+### Usage
+
+Each module described in the specification is to be one source file. All
+routines given in the specification for this module are to be in this file.
+
+
+### Auxiliary modules
```

Comment:
No mention of `submodules` should be extended

Reply:
Agreed. Submodules have now been recently added to our allowed code usage and this should reflect that.